lib/rspec/core/metadata_filter.rb

```diff
@@ -111,6 +111,10 @@ def prepend(item, metadata)
           @items_and_filters.unshift [item, metadata]
         end
 
+        def delete(item, metadata)
+          @items_and_filters.delete [item, metadata]
+        end
```

Comment:
This class and the `QueryOptimized` subclass are both directly unit tested in `filterable_item_repository_spec.rb`.  WDYT about adding specs for `delete` to them?